modules/apps/data-engine/data-engine-rest-impl/src/main/java/com/liferay/data/engine/rest/internal/dto/v2_0/util/DataDefinitionFieldUtil.java

```diff
@@ -308,16 +308,20 @@ else if (Objects.equals(
 			}
 		}
 
-		if (Validator.isNotNull(customProperties.get("ddmStructureId")) &&
-			Validator.isNull(customProperties.get("ddmStructureKey"))) {
+		String ddmStructureId = GetterUtil.getString(
+			customProperties.get("ddmStructureId"));
 
+		if (Validator.isNotNull(ddmStructureId)) {
 			DDMStructure ddmStructure =
 				ddmStructureLocalService.fetchDDMStructure(
-					GetterUtil.getLong(customProperties.get("ddmStructureId")));
+					GetterUtil.getLong(ddmStructureId));
 
 			if (ddmStructure != null) {
-				customProperties.put(
+				customProperties.putIfAbsent(
 					"ddmStructureKey", ddmStructure.getStructureKey());
+				customProperties.putIfAbsent(
+					"externalReferenceCode",
+					ddmStructure.getExternalReferenceCode());
```

Comment:
Hey @rijema 
I prefer to use the put method because this ensures that the code will always override this information with the data from the database (and we can trust 100% in this data).

Reply:
Okay!